dockstore-integration-testing/src/test/java/io/dockstore/client/cli/LimitedCRUDClientIT.java

```diff
@@ -232,6 +239,21 @@ public void testOverrideVersionLimit() throws IOException {
         api.editHostedTool(hostedTool.getId(), sourceFiles);
     }
 
+    @Test
+    public void testUploadZipHonorsVersionLimit() {
+        ApiClient webClient = BaseIT.getWebClient(BaseIT.ADMIN_USERNAME);
+        final HostedApi hostedApi = new HostedApi(webClient);
+        final Workflow hostedWorkflow = hostedApi.createHostedWorkflow("hosted", "something", "wdl", "something", null);
+        // Created workflow, no versions
+        File smartSeqFile = new File(ResourceHelpers.resourceFilePath("smartseq.zip"));
```

Comment:
Hmmm, does this mean https://github.com/ga4gh/dockstore/pull/2208/files#diff-0fe8aa76f8e46d26e28c81901f941045R218 isn't working for zip file uploads?

Reply:
Yes, it isn't working for zip file uploads.

It was always my intent, which I thought I had expressed in the design doc, but looking at it now, I didn't, that uploading a zip would be a complete version, rather than a merge with the previous version, like the existing PATCH. The thinking being that if somebody was posting a zip, they would want those exact set of files to be a version.

I think we should still do that, but should add a check for the versions being the same. If agreed, I can log that as a new issue.

Reply:
Agreed, new issue